### ISSUE TITLE
Lipid14 simulations

### ISSUE THREAD

**Issue by jmelcr** [NMRLipids/NmrLipidsCholXray#7]
Hi, is someone running simulations for Lipid14? 
There are [parameters for cholesterol](http://pubs.acs.org/doi/10.1021/acs.jpcb.5b04924) and the authors themselves made simulations with various cholesterol content - has anyone already asked them to share?

**Comment by ohsOllila** [comment]
I am going to contact them. They report order parameters in the publication and even compare to the same NMR dataset we are using. Those we could take directly from figures but I hope that they could give us, at least, numerical results for those. Form factors would be nice as well. Also if someone else has Lipid14 simulations with cholesterol, let us know.

**Comment by MykhailoGirych** [comment]
Hi, I'm going to convert Lipid14 parameters for cholesterol into Gromacs format and start the simulations in Sep. I will share the data as soon as I get it.

**Comment by jmelcr** [comment]
Good! I wanted to know whether there's something going on or whether we need to simulate it ourselves (which might still be fine to do for a check). 

@MykhailoGirych 
Could you share the parameters (files) with us even before the simulations are done, please?

**Comment by jjmadsen** [comment]
I have experimented with running lipid14 in amber before and can run some dynamics trajectories that are needed. 

@MykhailoGirych
How far did you get and which concentrations of interest are still missing?

**Comment by MykhailoGirych** [comment]
@jjmadsen The simulations in lipid14 are running (with the updated partial charges for POPC and CHOL taken from Lipid17). @jmelcr I'll share the parameters I use in a few days.

**Comment by jjmadsen** [comment]
@MykhailoGirych Great..Which concentrations are you running currently? I'd be happy to help out with this.

**Comment by jjmadsen** [comment]
@MykhailoGirych Hi, How's it going on this?

**Comment by ohsOllila** [comment]
@MykhailoGirych informed in Skype that he has the data, but he has not had time to upload it into Zenodo yet.

**Comment by jjmadsen** [comment]
Some issues have come to my attention regarding (my) lipid14 simulations. For completeness, I have been looking to reproduce findings described by Madej et al (2015). Briefly, I prepare systems with the CHARMM-GUI, equilibrate them, employ the charmmlipid2amber.py script to make a amber-compatible PDB structure, use tleap to make top+par files (w/ CHL.frcmod from Madej et al.), and finally use parmed to generate gromacs input files for the lipid14 simulations.  I think this approach should work just fine. 

But two potential issues have arisen.
1: The parameter modification files CHL.frcmod as taken line-by-line from the SI of Madej et al. ensures a Warning by my version of tleap (from amber18 tools); specifically: 
"Warning: Unexpected Improper Torsion PN term (-2)  Here is the Improper Torsion line in question:
cA cB cB cA   1           6.65         180.00          -2.00    SCEE=1.2 SCNB=2.0 Lipid11 v1.0 (GAFF c3-c2-c2-c3)"
2: Gromacs notices of lack of charge neutrality when running the system: "System total charge: -0.001"
To issue 2, this may simply be rounding because (for some reason) there is large variability in reported values wrt. significant figures in the CHL.frcmod file in the SI of Madej et al.

Has anyone else experienced this and may know what I could be doing wrong? The simulations themselves appear reasonable but, granted, I still have not calculated all observables and trajectories are not very long.

My pipeline for generating input files may affect what I observe; however, it would be very useful if I could compare topology files with someone that is also running these systems (successfully). @MykhailoGirych perhaps?

Thanks!

**Comment by batukav** [comment]
Hi,

I haven't worked with cholesterol in Amber, however I can try to reproduce the problem.

@jmelcr Could you please share CHL.frcmod, your input pdb and the leap script? I might be able to help, if @MykhailoGirych already doesn't have a solution.

**Comment by jmelcr** [comment]
> 2: Gromacs notices of lack of charge neutrality when running the system: "System total charge: -0.001"
To issue 2, this may simply be rounding because (for some reason) there is large variability in reported values wrt. significant figures in the CHL.frcmod file in the SI of Madej et al.

This is rather a common thing appearing when topologies are transformed - different round-up errors appearing with different number representation and truncation in the formats. A remedy is to calculate the total charge of the molecule (judging by your numbers, this will be roughly ~ -1.0 e-5) and add this value to the most burried atom in the molecule, so that the total charge of the molecule will become very close to zero ~ e-15.

> @jmelcr Could you please share CHL.frcmod, your input pdb and the leap script? I might be able to help, if @MykhailoGirych already doesn't have a solution.

I wish I had the topology - did not you mean @jjmadsen instead?

**Comment by batukav** [comment]
> I wish I had the topology - did not you mean @jjmadsen instead?

Indeed, sorry for the false tag.

**Comment by jjmadsen** [comment]
I have placed my example into a dropbox at this link: https://www.dropbox.com/s/pnop93doumhg0jn/lipid14_chl_test.tar.bz2?dl=0
out.pdb is the "amberfied" PDB, CHL.frcmod is reproduced from the SI of Madej et al. (manually; I check it three times, so it *should* be good - but no guarantees:), the rest is described in the readme and may be useful. The Setup is the of Madej et al. with lowest non-zero cholesterol and POPC (120/8=POPC/CHL; and other identical other parameters wrt. N_wat solvation=31/lipid, neutral system, 303K, etc)
@batukav @jmelcr

**Comment by batukav** [comment]
@jjmadsen Thank you for the files.

Regarding the improper dihedral warning:

I don't want to make a bold statement but it looks like there's an error in the CHL.frcmod file.

TL;DR
-2.00 value in the CHL.frcmod file for the improper torsion doesn't seem correct.

My long answer:

"1: The parameter modification files CHL.frcmod as taken line-by-line from the SI of Madej et al. ensures a Warning by my version of tleap (from amber18 tools); specifically:
"Warning: Unexpected Improper Torsion PN term (-2) Here is the Improper Torsion line in question:
cA cB cB cA 1 6.65 180.00 -2.00 SCEE=1.2 SCNB=2.0 Lipid11 v1.0 (GAFF c3-c2-c2-c3)"

-2.00 value (PN/periodicty) in the improper torsion term seems to be incorrect. I did some digging for this and found that the error is raised by amber.c file (./AmberTools/src/leap/src/leap/amber.c), line 668 in AmberTools2018. The relevant part is this:

```
static void
zAmberReadParmSetImpropers( PARMSET psParms, FILE *fIn )
{

STRING          sLine;
int             iRead, iN;
STRING          saStr[10];
double          dKp, dP0, dN, dScEE, dScNB;
BOOL            bPrintLine;

    memset(saStr, 0, sizeof(saStr));                    /* for Purify */
    while (1) {
        FGETS( sLine, fIn );

        NODASHES(sLine);
        /*
         *  get atoms & values, skipping possible 
         *      extraneous idivf value allowed by
         *      the input spec for PARM
         */
        iRead = sscanf( sLine, "%s %s %s %s",
                                saStr[0], saStr[1], saStr[2], saStr[3] );
        if ( iRead != 4 ) break;
        iRead = sscanf( sLine+15, "%lf %lf %lf", &dKp, &dP0, &dN );
        if ( iRead != 3 ) break;
        MESSAGE(( "Read: %s\n", sLine ));
        dN = (int)floor(dN+0.5);
        zAmberConvertWildCard( saStr[0] );
        zAmberConvertWildCard( saStr[1] );
        zAmberConvertWildCard( saStr[2] );
        zAmberConvertWildCard( saStr[3] );
        iN = (int)dN;
        dScEE = 0.0;
        dScNB = 0.0;
        /*
         *  check everything in case a format or other user error
         *      led to wrong values (e.g. IDIVF offset)
         */
        bPrintLine = FALSE;
        if ( dKp <= 0.0 ) {
            VPWARN(( "Expected Improper Torsion PK>0 (%f)\n", dKp ));
            bPrintLine = TRUE;
        }
        if ( dP0 < 179.999  ||  dP0 > 180.001 ) {
            VPWARN(( "Expected Improper Torsion PHASE=180 (%f)\n", dP0 ));
            bPrintLine = TRUE;
        }
        if ( iN < 1  ||  iN > 6 ) {
            VPWARN(( "Unexpected Improper Torsion PN term (%d)\n", iN ));
            bPrintLine = TRUE;
        }
        if ( bPrintLine )
            VP0(( "Here is the Improper Torsion line in question:\n%s",
                                                                sLine ));

        iParmSetAddImproperTerm( psParms,
                                saStr[0], saStr[1], saStr[2], saStr[3],
                                iN, dKp, dP0*DEGTORAD, dScEE, dScNB, "" );
    }
```
and it looks like the value -2.0 for PN is not allowed. 

I also checked the other frcmod/ff files and in none of them the improper dihedrals have a PN value less than 1.

This all makes me think that there's a typo/mistake in the original CHL.frcmod file (or I'm terribly wrong, that's also possible). 

What do you all think?

**Comment by jjmadsen** [comment]
@batukav It appears so. If this is intended, I'm not sure. I can add that there are dihedral entries with negative values for the PN periodicity term - and this is allowed. According to the manual for dihedral terms (but not improper dihedrals??) "NOTE: If PN .lt. 0.0 then the torsional potential is assumed to have more than one term, and the uses of the rest of the terms are read from the next cards until a positive PN is encountered.  The negative value of pn is used only for identifying the existence of the next term and only the absolute value of PN is kept. "  This makes me think that the sign is not simply a typo in the file, since there is a continuing line for the same four interaction partners in the CHL.frcmod file.  So, If there's a typo or if the file format expectations have changed over time, I cannot say.

**Comment by batukav** [comment]
@jjmadsen 
> "NOTE: If PN .lt. 0.0 then the torsional potential is assumed to have more than one term, and the uses of the rest of the terms are read from the next cards until a positive PN is encountered. The negative value of pn is used only for identifying the existence of the next term and only the absolute value of PN is kept. " This makes me think that the sign is not simply a typo in the file, since there is a continuing line for the same four interaction partners in the CHL.frcmod file.

This also could be the case. I checked the parm7 file generated by the leap using Parmed, and it looks like both improper torsion terms are correctly included in the parm7 file. So, -2.00 term in the improper torsions seems to follow the same procedure as the proper torsions only with an exception that we receive a warning in the case of improper torsions.

Well, in the end, the parm7 file seems to contain the right terms for the improper torsions so I think all good here. Yet, just to be sure we can also try dropping an email to the Amber mail list asking for a clarification.

**Comment by jmelcr** [comment]
@jjmadsen 
>  "NOTE: If PN .lt. 0.0 then the torsional potential is assumed to have more than one term, and the uses of the rest of the terms are read from the next cards until a positive PN is encountered. The negative value of pn is used only for identifying the existence of the next term and only the absolute value of PN is kept. " 

Multiple entries for potential contributions to dihedrals is indeed the case for Amber lipids. It looks to me from the code excerpt that even though a warning is raised, the line (with PN=-2) is used anyways. 

However, I do not see, why this is necessary - the two lines are the same but the PN number and force constant. Why use two, when one shall be sufficient? Maybe a typo in the topology?
`IMPROPER`
`cA-cB-cB-cA   1           6.65         180.00          -2.00	SCEE=1.2 SCNB=2.0 Lipid11 v1.0 (GAFF c3-c2-c2-c3)`
`cA-cB-cB-cA   1           1.90         180.00           1.00	SCEE=1.2 SCNB=2.0 Lipid11 v1.0 (GAFF c3-c2-c2-c3)`

If this feature, however, is rather important for the model (however I do not see a reason for that), this cannot be easily represented in Gromacs, which allows only a single entry per dihedral except the special type 9 proper dihedral. 

The choice of correct dihedral types for Amber Lipid force field in Gromacs topologies is is also discussed in [NMRlipdis project IV issue #12](https://github.com/NMRLipids/NMRlipidsIVPEandPG/issues/12#issuecomment-561098755). In short, the issue with Lipid17 POPG therein is likely to arise from using dihedral type 1 in the topology, which neglects all but the last entry for each dihedral in the topology file. 
**Dihedral type 9 shall be used instead also in the [above mentioned topology for Cholesterol](https://www.dropbox.com/s/pnop93doumhg0jn/lipid14_chl_test.tar.bz2?dl=0).** I have corrected the [cholesterol topology to use dihedral type 9 here](https://github.com/jmelcr/ecc_lipids/blob/master/topologies/lipid17/lipid14-CHOL_amber-order.top). The only issue (hopefully!) remains to solve the improper dihedrals, as there are multiple line entries and it is not very clear, whether they shall be used together (for which I do not see a reason), or is it a forgotten typo-mistake in the [SI of the Lipid14 cholesterol paper](https://pubs.acs.org/doi/suppl/10.1021/acs.jpcb.5b04924/suppl_file/jp5b04924_si_001.pdf).

**Comment by jjmadsen** [comment]
@jmelcr Thank you so much for your comments, which are very useful! I can try to find current contact information for the first author (Ben Madej) and reach out for final clarification that our assertions are correct.

**Comment by jjmadsen** [comment]
So, I have now been in contact with several people familiar with the lipid14 cholesterol model and can provide here some clarification. While Ben Madej and Ross Walker, who had not touched the project in a couple of years, could not from memory resolve the issue beyond what was described in the text and SI of their 2015 paper. However, Ross put me in contact with Callum Dickson, who is currently hands-on with related matters and he said that he too believed that the improper term of CHL.frcmod for lipid14 was actually a typo, as reported in the SI of the 2015 paper. In particular, the listed improper does not appear to be needed for the integrity of the cholesterol model (e.g., it is not present in the lipid17 model), and since the exact terms already appear in the DIHE dihedral section of the parameter modification file, it is most likely an copy-paste typo and we should be fine with ignoring the entry. Therefore, I propose that we are good to use the parameters I have put forward with the dihedral-type modification given by @jmelcr. I can run a test simulation using the corrected parameters and we can then compare with reported properties from Madej et al. in order to check whether they are accurately reproduced.

**Comment by jjmadsen** [comment]
This one got away from me, but I do have trajectories. I'm in the process of inspecting, processing, and getting it all uploaded to the repo in the near future.

**Comment by jjmadsen** [comment]
Here are the trajectories:

https://zenodo.org/record/3870998
https://zenodo.org/record/3871000
https://zenodo.org/record/3871004
https://zenodo.org/record/3871006
https://zenodo.org/record/3871008

I have not performed order parameter analysis yet

**Comment by ohsOllila** [comment]
The trajectories resulting from this discussion are included in the NMRlipids databank:
https://databank.nmrlipids.fi/trayectorias/85
https://databank.nmrlipids.fi/trayectorias/369
https://databank.nmrlipids.fi/trayectorias/421
https://databank.nmrlipids.fi/trayectorias/490
https://databank.nmrlipids.fi/trayectorias/566

On the other hand, Lipid14/17 simulations have become available in CHARMM-GUI meanwhile. Therefore, @mattijavanainen ran a new set of simulations with different box sizes to enable the PBC correction for lateral diffusion analysis for the NMRlipids III. These new simulations were also in the same temperature as the experiments, and are therefore used in the NMRlipids III project. Nevertheless, the conclusions are the same as in the simulations produced in here. I will close the issue now.
